internal/cortex/frontend/transport/utils/utils_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) The Cortex Authors.
+// Licensed under the Apache License 2.0.
+
+// Package utils Monitoring platform team helper resources for frontend
+package utils
+
+import (
+	"errors"
+	"testing"
+)
+
+// TestNewFailedQueryCache tests the NewFailedQueryCache function
+func TestNewFailedQueryCache(t *testing.T) {
+	cache, msg := NewFailedQueryCache(2)
+	if cache == nil {
+		t.Fatalf("Expected cache to be created, but got nil")
+	}
+	if msg != "" {
+		t.Fatalf("Expected no error message, but got: %s", msg)
+	}
+}
+
+// TestUpdateFailedQueryCache tests the UpdateFailedQueryCache method
+func TestUpdateFailedQueryCache(t *testing.T) {
+	cache, _ := NewFailedQueryCache(2)
+	lruCache := cache.LruCache
+
+	tests := []struct {
+		name                       string
+		err                        error
+		queryExpressionNormalized  string
+		queryExpressionRangeLength int
+		expectedResult             bool
+		expectedMessageContains    string
+	}{
+		{
+			name:                       "No error code in error message",
+			err:                        errors.New("no error code here"),
+			queryExpressionNormalized:  "test_query",
+			queryExpressionRangeLength: 60,
+			expectedResult:             false,
+			expectedMessageContains:    "msg: String regex conversion error, normalized query: test_query, query range seconds: 60, updating cache for error: no error code here",
+		},
+		{
+			name:                       "Non-cacheable error code",
+			err:                        errors.New("Code(500)"),
```

Comment:
Can you make the error string more like in real situation, e.g., "XXX Code(500) xxx"